weave/trace/settings.py

```diff
@@ -0,0 +1,72 @@
+import os
+from contextvars import ContextVar
+from typing import Any, Optional, Union
+
+from pydantic import BaseModel, ConfigDict, PrivateAttr
+
+disabled = ContextVar("disabled", default=False)
+print_call_link = ContextVar("disabled", default=True)
+
+
+class UserSettings(BaseModel):
```

Comment:
It seems like the order of precedent is:

ENV_VAR > init param > default

As such, i wonder if there is a more pluggable way to define settings.

For example, to add a setting right now, you need to:
1. Setup a context var
2. explicitly add a line to `def apply(self) -> None:`
3. add a method like `def should_disable_weave() -> bool:`, which need to also encode the env var.

I think an alternative approach would be something like:

```
class Setting():
   keyname: str
   default_value: any
   parse_env: Callable    # override to process env var


UserSettings.register(Setting("disabled", False))

# Then elsewhere

from weave.trace.settings import settings

settings.get("disabled")
```


this might make the system more maintainable

Reply:
add one line to add a setting `UserSettings.register(Setting("disabled", False))`, and let the system handle the rest

Reply:
How would a user pass in their own settings in this case?

Also it looks like `settings` is a dict or class, but that might not be threadsafe.

In general, I think we should keep it simple.  I've added helpers where they make sense and docs to make it more clear what a user needs to do.

Reply:
I'm still not convinced this is the best pattern to use It feels awkward to me that we have separate access methods for each setting which is divorced from the setting specification (here, represented as a pydantic model field). However, we can iterate - not worth to many back and forths here.